msl/equipment/resources/vaisala/ptb330.py

```diff
@@ -0,0 +1,72 @@
+"""
+Vaisala Barometer of type PTB330.
+The PTB330 barometer is available with one, two, or three barometer modules.
+"""
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from msl.equipment import EquipmentRecord
+from msl.equipment.resources import register
+
+from .ptu300 import PTU300
+
+
+@register(manufacturer=r'Vaisala', model=r'PTB330', flags=re.IGNORECASE)
+class PTB330(PTU300):
+
+    def __init__(self, record: EquipmentRecord) -> None:
+        """Vaisala Barometer PTB330 series.
+        Device manual is available `here <https://docs.vaisala.com/v/u/M210855EN-D/en-US>`_.
+
+        .. note::
+            Ensure the device is in STOP or SEND mode before initiating a connection to a PC.
+
+        Do not instantiate this class directly. Use the :meth:`~.EquipmentRecord.connect`
+        method to connect to the equipment.
+
+        :param record: A record from an :ref:`equipment-database`.
+        """
+        super(PTB330, self).__init__(record)
+
+    def set_units(self, desired_units: dict[str, str]) -> None:
+        """Set units of specified quantities.
+
+        :param desired_units: Dictionary of quantities (as keys) and their unit (their value)
+            as specified in the instrument manual on pages 15 and 73.
+
+            These may include the following (available options depend on the barometer components):
+
+                * Pressure quantities: P, P3h, P1, P2, QNH, QFE, HCP, ...
+                * Pressure units: hPa, psi, inHg, torr, bar, mbar, mmHg, kPa, Pa, mmH2O, inH2O
+                * Temperature quantities: T, TP1, TP2, TP3, ...
+                * Temperature units: 'C, 'F, or K
+        """
+        allowed_quantities_units = {}
+        self.write("UNIT ??")  # list the available measurement units for the quantities
+        for i in range(6 + len(self._pressure_modules)):
+            reply = self.read()
+            list_unit = [_u.strip() for _u in reply.split(":")]
+            allowed_quantities_units[list_unit[0]] = list_unit[1]
+
+        for quantity, u in desired_units.items():
+            if quantity not in allowed_quantities_units:
+                self.raise_exception(f'{quantity} is not a known quantity on this {self.info["Model"]} device. '
+                      f'Please select from {[i for i in allowed_quantities_units.keys()]}')
+            if u in allowed_quantities_units[quantity]:
+                reply = self.query(f"UNIT {quantity} {u}")
+                list_unit = [_u.strip() for _u in reply.split(":")]
+                self.units[list_unit[0]] = list_unit[1]
```

Comment:
You can use the unpacking feature of Python, rather than getting items by index later

```python
k, v = [_u.strip() for _u in reply.split(":")]
self._units[k] = v  # also note that I'm referencing the self._units (leading underscore) object to avoid multiple attribute lookups
```

Reply:
Ah, brilliant! I was thinking there would be a tidier way to do this :)